blivet/devices/partition.py

```diff
@@ -454,23 +454,22 @@ def _get_weight(self):
 
         # now we have the weights for varying mountpoints and fstypes by platform
         weight = 0
+
         if self.format.mountable and self.format.mountpoint == "/boot":
             weight = 2000
         elif (self.format.mountable and
               self.format.mountpoint == "/boot/efi" and
-              self.format.type in ("efi", "macefi") and
-              arch.is_efi()):
+              self.format.type in ("efi", "macefi")):
             weight = 5000
-        elif arch.is_x86() and self.format.type == "biosboot" and not arch.is_efi():
+        elif self.format.type == "biosboot":
```

Comment:
I think biosboot, appleboot and prepboot can be grouped here, just to simplify it a little bit more.